### PR TITLE
Add template for the upcoming migration script

### PR DESCRIPTION
**Pull request by sledges**
WIP

### CODE REVIEW

File: droid-hal-device.inc [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+# Device information:
 # device: should be the CM codename or the AOSP TARGET_PRODUCT
 # vendor: determine the droid-side directory used for ./device/<vendor>/<device>
 # device_pretty: User-visible model name of the device
 # vendor_pretty: User-visible manufacturer name of the device
 # rpm_device: device name used rpm-side (eg in configs) : defaults to device
 # rpm_vendor: vendor name used rpm-side (eg in configs) : defaults to vendor
+# Device capabilities:
+# installable_zip: adds zip updater scripts (default for all SFE devices)
+# enable_kernel_update: ability to flash kernel partition, inspect the usage of this macro
```
Comment (saukko) [inline droid-hal-device.inc]: Maybe add empty line with # here to categorize it a bit and same above.

File: helpers/build_packages.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+builddhd
+buildconfigs
+echo "-------------------------------------------------------------------------------"
+
+read -p 'About to perform "Build HA Middleware Packages" HADK chapter. Press Enter to continue.'
+sb2 -t $VENDOR-$DEVICE-armv7hl -R -msdk-install ssu domain sales
+sb2 -t $VENDOR-$DEVICE-armv7hl -R -msdk-install ssu dr sdk
+
+sb2 -t $VENDOR-$DEVICE-armv7hl -R -msdk-install zypper ref -f
+sb2 -t $VENDOR-$DEVICE-armv7hl -R -msdk-install zypper -n install droid-hal-$DEVICE-devel
```
Comment (saukko) [inline helpers/build_packages.sh]: Please don't hardcode armv7hl :)

File: helpers/util.sh [hunk 3]
```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
```
Comment (saukko) [inline helpers/util.sh]: Copyright missing.

File: helpers/util.sh [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        tail -n10 "$1"
+        minfo "Check `pwd`/`basename $1` for full log."
+    fi
+    shift
+    die $*
+}
+
+function buildconfigs() {
+    cd hybris/droid-configs
+    mb2 -t $VENDOR-$DEVICE-armv7hl \
```
Comment (saukko) [inline helpers/util.sh]: same here don't hardcode arch.

File: helpers/util.sh [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+source ~/.hadk.env
+
+ARCH="${ARCH:-armv7hl}"
```
Comment (sledges) [inline helpers/util.sh]: @saukko , how does that sound?

Comment (saukko) [inline helpers/util.sh]: Sure, eventually this should be removed when it hits the .hadk.env

**Comment by saukko** [comment]
lgtm

**Pull request merged**